### ISSUE TITLE
PEAR.WhiteSpace.ObjectOperatorIndent expects wrong indentation when there are multiple chained calls in the same statement

### ISSUE THREAD

**Issue by Daimona** [PHPCSStandards/PHP_CodeSniffer#1154]
NOTE: Copied over from https://github.com/squizlabs/PHP_CodeSniffer/issues/3718 with a reduced example and some more information.

## Describe the bug

If you have multiple chained calls within the same statement (for example, an array where multiple elements consist of chained method calls), but using different indentation, then the `PEAR.WhiteSpace.ObjectOperatorIndent` sniff will expect an incorrect amount of indentation.

### Code sample
```php
<?php

namespace Example;

function toJson(): array {
    return [
        'brand' => [
            'title' => $item->product()->brand(),
        ],
        'charges' => $item->charges()
            ->toArray(),
    ];
}
```

### Custom ruleset
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <rule ref="PEAR.WhiteSpace.ObjectOperatorIndent" />
</ruleset>
```

### To reproduce
1. Create a file called `test.php` with the code sample above
2. Run `phpcs test.php`
3. See error message displayed
```text
 11 | ERROR | [x] Object operator not indented correctly; expected 16 spaces but found 12
    |       |     (PEAR.WhiteSpace.ObjectOperatorIndent.Incorrect)
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
```

## Expected behavior

No issue emitted for the above code.

## Versions (please complete the following information)

|                                          |                                                                                                                |
|-----------------------------|---------------------------------------------------------------------------- |
| Operating System            | Ubuntu 24.04                                                           |
| PHP version                      | 8.3                                                                                          |
| PHP_CodeSniffer version | master                                                                                  |
| Standard                          | custom                                                          |
| Install type                       | composer |

## Additional context

The sniff does: https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/368817d890b7b872b39ff188f739434244bb240c/src/Standards/PEAR/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php#L76-L77

But the whole array literal in the above example is just one statement. So, when analyzing the chained calls at line 8, `$next` will find the chained calls at line 10-11 and expect them to be indented like the ones at line 8.

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by rodrigoprimo** [comment]
Thanks for copying this issue from the old repository, @Daimona. Also, thanks to @lucassabreu for the original report.

I was able to reproduce it. Here is a simplified code snippet that produces the same bug:

```php
$array = [
    [
        $item->first()->second(),
    ],
    $item->first()
        ->second(),
];
```

> But the whole array literal in the above example is just one statement. So, when analyzing the chained calls at line 8, $next will find the chained calls at line 10-11 and expect them to be indented like the ones at line 8.

I believe the problem is actually in another part of the code. The call to `File::findEndOfStatement()` correctly identifies the comma as the end of the statement and assigns its position to `$end`. `$end` is used to limit the scope of the search performed by `File::fileNext()`, and thus it does not include the chained calls on lines 10-11.

Below, the sniff loops over all the object operators in a given chained call using a `while` loop:

https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/368817d890b7b872b39ff188f739434244bb240c/src/Standards/PEAR/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php#L118

Here is the code where `$next` is reassigned to the next object operator:

https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/368817d890b7b872b39ff188f739434244bb240c/src/Standards/PEAR/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php#L191-L198

I believe the problem is that the sniff is relying on the `$local` parameter of `findNext()` to limit the scope of the search method, and `$local` only considers a semicolon as the end of a statement:

https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/2e9da03b4ce03b741bd30fdf83f5e2eb1f55b1c8/src/Files/File.php#L2422-L2424

Unless I'm missing something, this is a relatively easy fix to make, and I will prepare a PR for it.
